shims/spark32/src/main/scala/org/apache/spark/sql/execution/FileSourceScanExecShim.scala

```diff
@@ -53,6 +53,20 @@ class FileSourceScanExecShim(
   // Note: "metrics" is made transient to avoid sending driver-side metrics to tasks.
   @transient override lazy val metrics: Map[String, SQLMetric] = Map()
 
+  val unsupportedColumns: Seq[String] = {
```

Comment:
FileSourceScan targets on datasource v1, do we have similar issue for datasource v2? If so, can this part of logic be put in `BasicScanExecTransformer`? Maybe by adding a function receives output as parameter.